libs/api/domains/municipalities-financial-aid/src/lib/applicationApiProvider.ts

```diff
@@ -0,0 +1,22 @@
+import { Provider } from '@nestjs/common/interfaces/modules/provider.interface'
+
+import { ConfigType } from '@island.is/nest/config'
+
+import {
+  ApplicationApi,
+  Configuration,
+} from '@island.is/clients/municipalities-financial-aid'
+import { MunicipalitiesFinancialAidConfig } from './municipalitiesFInancialAid.config'
+
+export const ApplicationApiProvider: Provider<ApplicationApi> = {
```

Comment:
This should be exported from the client library. The config should be there as well.

The domain (and other modules) can then import the client library module, and get ApplicationAPI injected wherever.

If you want, the client library could also have its own service class if you want to share some “integration” logic between your projects. E.g. 404 handling. Just be mindful that it doesn’t get too coupled to one of the apps.